src/il_representations/scripts/interpret.py

```diff
@@ -208,161 +136,71 @@ def deep_lift(net, tensor_image, label):
-    y_axis_lc_attr_test = lc_attr_test.mean(0).detach().numpy()
-    y_axis_lc_attr_test = y_axis_lc_attr_test / np.linalg.norm(y_axis_lc_attr_test, ord=1)
-
-    y_axis_lin_weight = layer_weight[0].detach().numpy()
-    y_axis_lin_weight = y_axis_lin_weight / np.linalg.norm(y_axis_lin_weight, ord=1)
-
-    width = 0.25
-    legends = ['Attributions', 'Weights']
-    x_axis_labels = ['Neuron {}'.format(i) for i in range(len(y_axis_lin_weight))]
-
-    ax = plt.subplot()
-    ax.set_title('Aggregated neuron importances and learned weights in the indicated linear layer of the model')
-
-    ax.bar(x_axis_data + width, y_axis_lc_attr_test, width, align='center', alpha=0.5, color='red')
-    ax.bar(x_axis_data + 2 * width, y_axis_lin_weight, width, align='center', alpha=0.5, color='green')
-    plt.legend(legends, loc=2, prop={'size': 20})
-    ax.autoscale_view()
-    plt.tight_layout()
-
-    ax.set_xticks(x_axis_data + 0.5)
-    ax.set_xticklabels(x_axis_labels)
-
-    if save_dir:
-        plt.savefig(f'{save_dir}/{save_name}.png')
-        print(f'Saved image at {save_dir}/{save_name}.png')
-    if show_imgs:
-        plt.show()
-
-
-def choose_layer(network, module_name, layer_idx):
-    """
-    network - The network to be interpreted.
-    module_name - The network module being interpreted. It should be either "encoder" or "decoder".
-    layer_idx - The index of the layer to be studied. It belongs to the module specified by module_name.
+@interp_ex.capture
+def prepare_network(combined_meta, encoder_path, device):
+    policy = make_policy(encoder_path=encoder_path,
+                         policy_continue_path=None,
+                         observation_space=combined_meta['observation_space'],
+                         action_space=combined_meta['action_space'],
+                         ortho_init=False,
+                         log_std_init=0.0,
+                         postproc_arch=(),
+                         freeze_pol_encoder=True,
+                         encoder_kwargs={},
+                         algo='bc',
+                         lr_schedule=None)
+    # TODO: make this from action space
+    return LoadedPolicy(policy, use_first_action=False).to(device)
+
+
+class LoadedPolicy(nn.Module):
+    """A wrapper class for loaded policy so it has the same interface as required
+    by Captum. Specifically, most of our saved policies are ActorCriticCnnPolicy,
+    and its forward() function outputs a tuple of (actions, values, log_prob).
+    In Captum we only need its actions.
     """
+    def __init__(self, policy, use_first_action):
+        super().__init__()
+        self.policy = policy
+        self.use_first_action = use_first_action
 
-    if module_name == 'encoder':
-        rep_encoder = network.policy.features_extractor.representation_encoder
+    def forward(self, x):
+        # TODO: make this compatible with both DMC and procgen.
+        action = self.policy.action_net(self.policy.features_extractor(x/255.))
```

Comment:
Maybe add a comment to clarify what `action_net` produces? Originally I assumed it would produce an action (which is a problem for procgen, since you can't backpropagate through the discrete action sampling mechanism). On reading the SB3 source, though, it seems like it either produces (1) logits for a categorical distribution, or (2) a mean vector for a Gaussian distribution. Writing this on the line above would be useful to future readers.